ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -540,6 +548,64 @@ private EndpointResponse handleScalablePushQuery(
     return EndpointResponse.ok(queryStreamWriter);
   }
 
+  private EndpointResponse handleStreamPullQuery(
+      final ImmutableAnalysis analysis,
+      final ServiceContext serviceContext,
+      final ConfiguredStatement<Query> configured,
+      final CompletableFuture<Void> connectionClosedFuture) {
+
+    final StreamPullQueryMetadata streamPullQueryMetadata = ksqlEngine
+        .createStreamPullQuery(
+            serviceContext,
+            analysis,
+            configured,
+            false
+        );
+    localCommands.ifPresent(lc -> lc.write(streamPullQueryMetadata.getTransientQueryMetadata()));
+
+    final QueryStreamWriter queryStreamWriter = new QueryStreamWriter(
+        streamPullQueryMetadata.getTransientQueryMetadata(),
+        disconnectCheckInterval.toMillis(),
+        OBJECT_MAPPER,
+        connectionClosedFuture,
+        new StreamPullQueryCompletionCheck(ksqlEngine, streamPullQueryMetadata)
+    );
+
+    return EndpointResponse.ok(queryStreamWriter);
+  }
+
+  private static final class StreamPullQueryCompletionCheck implements Supplier<Boolean> {
```

Comment:
Added some extra logic here so that once we're "complete", we continue to report "true" without any more API calls. And also so that we won't check more than every 100ms.

Reply:
I suspect this will be something that we will want to factor out to the other endpoints in some manner if we continue to use roughly this approach with the others.

That said, this does a blocking call to kafka, so you probably will have to wrap this in a runnable to put on the blocking worker pool, since `/query-stream` doesn't allow blocking calls.  Not hard, but just wanting to point that out.